src/NRedisStack/Search/Schema.cs

```diff
@@ -275,25 +278,340 @@ public VectorField(FieldName name, VectorAlgo algorithm, Dictionary<string, obje
+                + (Type == VectorType.NotSpecified ? 0 : 1)
+                + (DistanceMetric == VectorDistanceMetric.NotSpecified ? 0 : 1);
+
+            internal virtual void AddDirectAttributes(List<object> args)
+            {
+                if (Dimensions != 0)
+                {
+                    args.Add("DIM");
+                    args.Add(Dimensions);
+                }
+                if (Type != VectorType.NotSpecified)
+                {
+                    args.Add("TYPE");
+                    args.Add(Type switch
+                    {
+                        VectorType.FLOAT32 => "FLOAT32",
+                        VectorType.FLOAT64 => "FLOAT64",
+                        VectorType.BFLOAT16 => "BFLOAT16",
+                        VectorType.FLOAT16 => "FLOAT16",
+                        _ => Type.ToString(), // fallback
+                    });
+                }
+
+                if (DistanceMetric != VectorDistanceMetric.NotSpecified)
+                {
+                    args.Add("DISTANCE_METRIC");
+                    args.Add(DistanceMetric switch
+                    {
+                        VectorDistanceMetric.EuclideanDistance => "L2",
+                        VectorDistanceMetric.InnerProduct => "IP",
+                        VectorDistanceMetric.CosineDistance => "COSINE",
+                        _ => DistanceMetric.ToString(), // fallback
+                    });
+                }
+            }
+
+            /// <summary>
+            /// The storage type for a vector field.
+            /// </summary>
+            public enum VectorType
+            {
+                /// <summary>
+                /// Not specified, or specified via <see cref="VectorField.Attributes"/>.
+                /// </summary>
+                NotSpecified = 0,
+                /// <summary>
+                /// 32-bit floating point.
+                /// </summary>
+                FLOAT32 = 1,
+                /// <summary>
+                /// 64-bit floating point.
+                /// </summary>
+                FLOAT64 = 2,
+                /// <summary>
+                /// 16-bit "brain" floating point
+                /// </summary>
+                BFLOAT16 = 3, // requires v2.10 or later.
+                /// <summary>
+                /// 16-bit floating point.
+                /// </summary>
+                FLOAT16 = 4, // requires v2.10 or later.
+            }
+
+            /// <summary>
+            /// The distance metric to use for vector similarity search.
+            /// </summary>
+            public enum VectorDistanceMetric
+            {
+                /// <summary>
+                /// Not specified, or specified via <see cref="VectorField.Attributes"/>.
+                /// </summary>
+                NotSpecified = 0,
+                /// <summary>
+                /// Euclidean distance between two vectors.
+                /// </summary>
+                EuclideanDistance = 1,
```

Comment:
Please update the doc strings to refer to Redis "raw" names.

Reply:
done